proposals/slottable-request.md

```diff
@@ -0,0 +1,86 @@
+# Slottable Request Protocol
+aka "Render Props for Web Components"
+
+**Author**: Kevin Schaaf (@kevinpschaaf, Google)
+
+**Status**: Draft
+
+**Last Updated**: 2023-08-28
+
+
+# Background
+
+There are often use cases that require a component to render UI based on data, but where the specific rendering of the data should be controllable by the user. Use cases of this include:
+
+*   A data table or tree component, which manages the rendering of cells or tree nodes, but where the given look and feel of the items is customizable by the user.
+*   A virtual list component, which optimizes rendering a list of data by only rendering a subset of user-templated items based on which array instances are visible on the screen.
+*   Components that fetch data themselves, but provide the ability for the user to customize the rendering of the data.
+
+This proposal describes an interoperable protocol for components to request that their owning context render slotted content into themselves, using data provided via the protocol.
+
+# Goals
+
+*   The ability for users of a component to render requested content using a templating system of their choice, using data supplied by the component.
+*   The ability for requested content to be logically composed by the requesting component into the component's shadow root at a position of its choice.
```

Comment:
This one doesn't read very clearly to me. Do you meant that the rendered content can have slot attributes?

Reply:
It's just trying to say that, while the provider gets to provide the content, the requestor gets control over "where" it goes.

Slots is how we fulfill the requirement in the protocol, but this requirement alone could be satisfied by e.g. providing a function that returns DOM (but that wouldn't meet the other requirements). 

Maybe this is clearer?
```suggestion
*   The ability for requested content to be rendered into the requesting component's shadow root at a position of its choice.
```

Reply:
Reworded.